### ISSUE TITLE
Additional network configuration for wpa_supplicant.conf ?

### ISSUE THREAD

**Issue by geronime** [gisogrimm/ovbox#3]
Hello and first of all many thanks for this great tool!

I've successfully used ovbox macos desktop application and now I wanted to set up a Raspberry Pi version. However because of the layout of my apartment, I would really need a WLAN connection (which I already successfully tested using the MacOS desktop app and playing with other two persons was just fine).

Having played around with the Pi version, I discovered the undocumented `/boot/ovclient-wifi.txt` file which is needed to have the wlan interface enabled and causes a reset of `wpa_supplicant` configuration on each boot through the `autorun` script by calling non-interactive `raspi-config`.

After a long testing I found out that I am only able to connect to my WLAN if I add the `key_mgmt` configuration of the `network` block in the `wpa_supplicant.conf` file (I have relatively modern WLAN router with WPA3).

My understanding is that the `/boot/ovclient-wifi.txt` is a nice way on how to easily change WLAN configuration by simply modifying the file directly on the SD card on any device with a microSD card reader (potentially even an Android phone), but for my home WLAN network it is counterproductive and actually prevents me from connecting the ovbox Pi to my WLAN.

I am not sure how to solve this systematically, but a workaround could be to comment out the [rfkill block all](https://github.com/gisogrimm/ovbox/blob/master/tools/pi/autorun#L32) (or replace `all` with `bluetooth`) and then I should be able to remove the `/boot/ovclient-wifi.txt` file and stop having my extended wpa_supplicant configuration replaced on each startup. So far I haven't been able to figure out how to make permanent changes to the `pi`'s `autorun` script so that it's not refreshed on each start.

What is your experience with ovbox using Raspberry PI and WLAN?

**Comment by gisogrimm** [comment]
Hello geronime,
thanks for reporting this. Unfortunately I tested this only with WPA2 routers. The `/boot/ovclient-wifi.txt` can actually be modified via the configuration webinterface ("device settings", activate "expert settings", see "WiFi settings" towards the lower end of the page - please note that using the webinterface would store your WLAN password in clear text on the configuration server).
If the WLAN connection is not needed then disabling the radio potentially improves latency, which is the reason for disabling it at boot time. I will look into a solution in the next months (I plan to create a new installer image based on more recent and also arm64 RaspiOS anyway).
As a workaround you may edit the autorun script `/home/pi/autorun` - you need to disable the overlayfs first with `sudo raspi-config nonint disable_overlayfs`.

In general WLAN can work well if the network quality is very high and no competing networks are around. Any type of radio based connection will result in a larger jitter (due to the way package collisions and errors are handled), this unfortunately also includes WLAN.

**Comment by geronime** [comment]
> The /boot/ovclient-wifi.txt can actually be modified via the configuration webinterface ("device settings", activate "expert settings", see "WiFi settings" towards the lower end of the page - please note that using the webinterface would store your WLAN password in clear text on the configuration server).

Ah, that's clever, I was manually modifying the file directly on the SD card. Still it doesn't allow me to add additional network parameters such as `key_mgmt=WPA-PSK-SHA256` as I am not really sure it is even possible to pass it to the [raspi-config noint do_wifi_ssid_passphrase](https://github.com/gisogrimm/ovbox/blob/master/tools/pi/autorun#L69) command.

> I plan to create a new installer image based on more recent and also arm64 RaspiOS anyway

Perfect, thanks for that! And thanks for your effort on this project, it actually enabled me to restart playing semi-regularly after our local group disbanded because of the pandemic.

> As a workaround you may edit the autorun script `/home/pi/autorun` - you need to disable the overlayfs first with `sudo raspi-config nonint disable_overlayfs`.

That was the plan but until now I haven't had the time to figure out how to make my changes to the script durable. This is exactly what I need. Thanks!

> In general WLAN can work well if the network quality is very high and no competing networks are around. Any type of radio based connection will result in a larger jitter (due to the way package collisions and errors are handled), this unfortunately also includes WLAN.

I fully understand and accept that. Having had a rehearsal with other two persons, they haven't noticed any issues, except one time when my shared upload was not up to the job of sending 44.1kHz sampling rate signal, as I wasn't able to configure it lower on macos (for lower sampling rate the web interface started showing an error message complaining about jack daemon crashing or something of that sort). That stuttering was most likely caused by my internet connection and not by the home network.

Thanks for the guidance, feel free to close the issue, I can take it from here to adjust the autostart script locally.

**Comment by geronime** [comment]
Okay I successfully managed to get the OVbox working on Pi over WLAN, thanks for the advice on how to disable the overlay FS. However, when testing in the Echo room, I get quite a bit of audible noise using the official Pi power adapter. When I supplied the Pi from a power bank, there's no audible noise. My previous two OVbox sessions using MacBook desktop app (over WLAN) were also without noticeable noise and now I tested that after connecting the MacBook to the power adapter while in the Echo room, I don't get additional noise I could hear. I noticed other people complaining about the noisy Pi power adapter. What's your experience with noise when using Pi?

**Comment by gisogrimm** [comment]
This is not related to WLAN: the USB power supply of the Raspberry Pi is not of the best quality, which will result in noise on some audio interfaces. The Focusrite Scarlett solo or 2i2 should work without noise. Some audio interfaces require an external power supply even when they work without on PC or MacBook.

**Comment by geronime** [comment]
Yes, I understood it's not related to WLAN - all my tests were using WLAN. I have Scarlett Solo and with Pi there's audible noise when using the Pi power supply compared to using a power bank. 

The noise surprised me during my initial Echo room test, because I couldn't hear such a noise during my previous OVbox sessions using a MacOS desktop app (the same setup and the same mic/output volumes on the Scarlett, only a MacBook instead of the Pi).

I was curious if you (or anybody else) have experience with using some less noisier power supply like iFi iPower2 or Allo Nirvana in their OVbox@Pi setup to see whether a ~70€ PSU is worth considering.

**Comment by gisogrimm** [comment]
If you hear your monitor signal clearly without noise (e.g. when you enter an empty room) than the noise could be caused by dropouts. In that case increasing the jitter buffer may solve the issue. However, I do not understand how this relates to the power supply.
The only noise problems with Raspberry Pi and Scarlett solo I know of were related to powerline ethernet. I am using the original Raspberry Pi USB C power supply.

**Comment by geronime** [comment]
It's hard to explain. On entering an empty room I hear a subtle noise when having Pi powered by the original power supply. When I supplied the Pi with a power bank, I couldn't hear any - all with the same settings, the only changed element was the source of power for the Pi. It's possible this is highlighted for me because I use my (quite old but still gold) AKG K812 headphones and there's no headamp in this setup. I have run into people asking about less noisier power supplies for the Pi, for example [here](https://community.roonlabs.com/t/raspberry-pi-4b-ultra-low-noise-power-supply/171665).

**Comment by geronime** [comment]
In case anybody else finds this thread: I have tested iFi iPower2 5V/2.5A power adapter which turned out to be sufficient to supply a Raspberry Pi 4B and Scarlett Solo. There is noticeably lower noise once entering an empty room compared to the stock unit, however for me the noise reduction is not significant enough to warrant a 69 EUR purchase. I ended up using my power bank which is able to provide for a several hour noise-free OVbox session.
